scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetService.java

```
@@ -0,0 +1,220 @@
+        try {
+            fileSystemService.addDatasetDirectories(String.valueOf(savedDataset.getId()),
+                    user.getUsername());
+        } catch (DatasetStorageException e) {
+            throw new DatasetStorageException(
+                    "An IOException prevented the Dataset from being created", e);
+        }
+    }
+
+    /**
+     * Edits the dataset name in the database (Dataset path uses id).
+     *
+     * @param id      The id of the dataset to edit.
+     * @param newName The new name of the dataset being edited.
+     */
+    @Override
+    public void editDataset(Long id, String newName) {
+
+        // filesystem path is: username/datasetId so no change is needed to the filesystem
+        Optional<Dataset> dataset = datasetRepository.findById(id);
+        if (dataset.isPresent()) {
+            dataset.get().setName(newName);
+            datasetRepository.save(dataset.get());
+        }
+    }
+
+    /**
+     * Deletes the dataset in the filesystem, then in the database.
+     *
+     * @param id       The id of the dataset to delete.
+     * @param username The username of the owner of the Dataset
+     * @throws DatasetStorageException If the Dataset couldn't be deleted
+     */
+    @Override
+    public void deleteDataset(Long id, String username) throws DatasetStorageException {
+        // Delete Dataset Directories and files from the filesystem
+        try {
+            fileSystemService.deleteDatasetDirectories(id, username);
+        } catch (SecurityException e) {
+            throw new DatasetStorageException("Read access to directory was denied", e);
+        }
+        // Then delete Dataset and its files from the database
+        Optional<Dataset> dataset = datasetRepository.findById(id);
+        if (dataset.isPresent()) {
+            dataFileRepository.deleteAllById(id);
+        } else {
+            throw new DatasetStorageException("Dataset removed from filesystem not found in database");
```

I don't think that trying to delete a non-existing element warrants an exception. I also dont think that spring minds, so I wouldn't throw an exception here but just silently go on (the element you are trying to delete doesn't exist, so you end up with the same result as if it would exist).